### PR TITLE
Add notes about python udf sdk update

### LINKED ISSUE #2162: Document: refactor(udf): switch to the latest arrow-udf versions
This issue tracks the documentation update needed for the merged PR #16619.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/16619
Source PR Merged At: 2024-05-11T04:25:25Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/16619

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2162

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Comment by wangrunji0408** [comment]
I have added the migration guide for upgrading. Please help review. Thanks! @ShanlanLi

File: docs/sql/udf/udf-python-embedded.md [hunk 1]
```diff
@@ -38,6 +38,10 @@ The Python code must contain a function that has the same name as declared in th
 
 See the correspondence between SQL types and Python types in the [Data type mapping](udf-python-embedded.md#data-type-mapping).
 
+:::warn
+Due to the nature of Python, the correctness of the source code cannot be verified when creating a function. It is recommended to make sure your implementation is correct through batch query before using UDFs in materialized views. If an error occurs when executing UDF in materialized views, you will see that the output results are all NULL.
+:::
```
Comment (wangrunji0408) [inline docs/sql/udf/udf-python-embedded.md]: This section is not related to the upgrading. I just added it by the way.

File: docs/sql/udf/udf-python.md [hunk 2]
```diff
@@ -91,16 +97,13 @@ def blocking(x):
...
-    src_addr, dst_addr = struct.unpack('!4s4s', tcp_packet[12:20])
-    src_port, dst_port = struct.unpack('!HH', tcp_packet[20:24])
-    src_addr = socket.inet_ntoa(src_addr)
-    dst_addr = socket.inet_ntoa(dst_addr)
-    return src_addr, dst_addr, src_port, dst_port
+# Define a scalar function that returns multiple values within a struct
+@udf(input_types=['VARCHAR'], result_type='STRUCT<key: VARCHAR, value: VARCHAR>')
+def key_value(pair: str):
+    key, value = pair.split('=')
+    return {'key': key, 'value': value}
```
Comment (wangrunji0408) [inline docs/sql/udf/udf-python.md:104]: Changed the example of struct type to focus more on interface rather than implementation.

File: docs/sql/udf/udf-python-embedded.md [hunk 3]
```diff
@@ -38,6 +38,10 @@ The Python code must contain a function that has the same name as declared in th
 
 See the correspondence between SQL types and Python types in the [Data type mapping](udf-python-embedded.md#data-type-mapping).
 
+:::warn
```
Comment (emile-00) [inline docs/sql/udf/udf-python-embedded.md]:
```suggestion
:::warning
```

File: docs/sql/udf/udf-python.md [hunk 4]
```diff
@@ -261,7 +253,41 @@ The RisingWave Python UDF SDK supports the following data types:
...
+
+The type aliases `FLOAT4` and `FLOAT8` are removed and replaced by `REAL` and `DOUBLE PRECISION`.
+
+```diff
+- @udf(input_types=['FLOAT4', 'FLOAT8'], result_type='INT')
++ @udf(input_types=['REAL', 'DOUBLE PRECISION'], result_type='INT')
+```
+
+The `STRUCT` type now requires field names. The field names must exactly match the ones defined in `CREATE FUNCTION`.
+The function that returns a struct type now returns a dict instead of a tuple. The field names of the dictionary must match the definition in the signature.
```
Comment (emile-00) [inline docs/sql/udf/udf-python.md]:
```suggestion
The function that returns a struct type now returns a dictionary instead of a tuple. The field names of the dictionary must match the definition in the signature.
```

**Review by emile-00 — APPROVED** [review]
Left some small suggestions. Otherwise LGTM!

File: docs/sql/udf/udf-python-embedded.md [hunk 5]
```diff
@@ -38,6 +38,10 @@ The Python code must contain a function that has the same name as declared in th
 
 See the correspondence between SQL types and Python types in the [Data type mapping](udf-python-embedded.md#data-type-mapping).
 
+:::note
+Due to the nature of Python, the correctness of the source code cannot be verified when creating a function. It is recommended to make sure your implementation is correct through batch query before using UDFs in materialized views. If an error occurs when executing UDF in materialized views, you will see that the output results are all NULL.
```
Comment (WanYixian) [inline docs/sql/udf/udf-python-embedded.md]:
```suggestion
Due to the nature of Python, the correctness of the source code cannot be verified when creating a function. It is recommended to make sure your implementation is correct through batch query before using UDFs in materialized views. If an error occurs when executing UDF in materialized views, all output results will be NULL.
```

File: docs/sql/udf/udf-python.md [hunk 6]
```diff
@@ -21,14 +21,22 @@ This article provides a step-by-step guide for installing the RisingWave UDF API
...
+pip install arrow-udf
 ```
 
 <details>
 <summary>Cannot run this command?</summary>
 If "command not found: pip" is returned, <a href="https://packaging.python.org/en/latest/tutorials/installing-packages/#ensure-you-can-run-pip-from-the-command-line">check if pip is available</a> in your environment and <a href="https://packaging.python.org/en/latest/tutorials/installing-packages/#ensure-pip-setuptools-and-wheel-are-up-to-date">ensure it is up to date</a>.
 </details>
 
+:::note
+The current Python UDF SDK is supported since version 1.10 and is not supported in older versions.
```
Comment (WanYixian) [inline docs/sql/udf/udf-python.md:33]:
```suggestion
The current Python UDF SDK is supported since version 1.10.
```

**Review by WanYixian — APPROVED** [review]
Others lgtm
